kyaml/fn/framework/example_filter_GVK_test.go

```diff
@@ -0,0 +1,34 @@
+package framework_test
+
+import (
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+)
+
+// This example implements a function that reads the desired replicas from the
+// functionConfig and updates the replicas field for all deployments.
+
+func Example_filterGVK() {
```

Comment:
Just so we're all on the same page about what's already possible, an API-oriented version of the same thing would look like the code below. Personally I prefer to write my functions this way, because it is only slightly more code and it is much clearer about the function's API, validation rules and processing steps.

```go
func Example_filterGVK() {
	if err := command.AsMain(&ReplicaUpdater{}); err != nil {
		os.Exit(1)
	}
}

type ReplicaUpdater struct {
	Replicas *int `yaml:"replicas" json:"replicas"`
}

func (u *ReplicaUpdater) Validate() error {
	if u.Replicas == nil {
		return framework.ErrMissingFnConfig{RequiredPaths: []string{".Replicas"}}
	}
	return nil
}

func (u *ReplicaUpdater) Process(rl *framework.ResourceList) error {
	if err := framework.LoadFunctionConfig(rl.FunctionConfig, u); err != nil {
		return errors.Wrap(err)
	}
	for i := range rl.Items {
		if framework.GVKMatcher("apps/v1/Deployment").Match(rl.Items[i]) {
			rl.Items[i].SetOrDie(u.Replicas, "spec", "replicas")
		}
	}
	return nil
}
```

And since this processor doesn't do anything special outside the filtering logic, here's the same thing again using `SimpleProcessor`: 

```go

func Example_filterGVK() {
	updater := &ReplicaUpdater{}
	if err := command.AsMain(framework.SimpleProcessor{Config: updater, Filter: updater}); err != nil {
		os.Exit(1)
	}
}

type ReplicaUpdater struct {
	Replicas *int `yaml:"replicas" json:"replicas"`
}

func (u *ReplicaUpdater) Validate() error {
	if u.Replicas == nil {
		return framework.ErrMissingFnConfig{RequiredPaths: []string{".Replicas"}}
	}
	return nil
}

func (u *ReplicaUpdater) Filter(items []*yaml.RNode) ([]*yaml.RNode, error) {
	for i := range items {
		if framework.GVKMatcher("apps/v1/Deployment").Match(items[i]) {
			items[i].SetOrDie(u.Replicas, "spec", "replicas")
		}
	}
	return items, nil
}
```

Writing that out, it is a little weird to have to provide the `updater` to both fields... maybe a nice optimization could be that if a Config is provided, a Filter isn't, and Config _is_ a filter, we use it automatically. WDYT about that enhancement?